### PR TITLE
Cb 5562 Fix bad this.setState for the new form API

### PR DESCRIPTION
**Pull request by sergeyteleshev**

### CODE REVIEW

File: webapp/packages/core-ui/src/Form/FormState.ts [hunk 1]
```diff
@@ -25,16 +25,14 @@ export class FormState<TState> implements IFormState<TState> {
   mode: FormMode;
   parts: MetadataMap<string, IFormPart<any>>;
   state: TState;
-  isSaving: boolean;
 
   statusMessage: string | string[] | null;
   statusType: ENotificationType | null;
-  exception: Error | (Error | null)[] | null;
```
Comment (Wroud) [inline webapp/packages/core-ui/src/Form/FormState.ts:32]: it's better to keep it as a getter like with `isDisabled`

File: webapp/packages/core-ui/src/Form/FormState.ts [hunk 2]
```diff
@@ -170,10 +151,6 @@ export class FormState<TState> implements IFormState<TState> {
         }
 
         await this.fillDefaultConfigTask.execute(this);
-        this.exception = null;
-      } catch (exception: any) {
-        this.exception = exception;
-        throw exception;
       } finally {
         this.promise = null;
```
Comment (Wroud) [inline webapp/packages/core-ui/src/Form/FormState.ts]: `load` method should be removed together with `configureTask`

**Review by Wroud — CHANGES_REQUESTED** [review]

File: webapp/packages/core-ui/src/Form/FormState.ts [hunk 3]
```diff
@@ -90,42 +86,36 @@ export class FormState<TState> implements IFormState<TState> {
...
+      .map(part => part?.exception)
+      .flat()
+      .filter(isNotNullDefined);
   }
 
-  isError(): boolean {
-    return this.dataContext.get(DATA_CONTEXT_LOADABLE_STATE)!.loaders.some(loader => loader.isError());
+  get isError(): boolean {
+    return Array.from(this.parts.values()).some(part => part.isError());
   }
```
Comment (Wroud) [inline webapp/packages/core-ui/src/Form/FormState.ts]: for optimization you better to create private computed getter for part's value `return Array.from(this.parts.values())` and reuse it in other computeds, but don't forget to configure proper `equals` method

example:
```
parts: computed<type[]>({
        equals: isArraysEqual,
      }),
```

File: webapp/packages/core-ui/src/Form/FormState.ts [hunk 4]
```diff
@@ -210,32 +156,22 @@ export class FormState<TState> implements IFormState<TState> {
...
       }
 
-      this.exception = null;
       return true;
-    } catch (exception: any) {
-      this.exception = exception;
-    } finally {
-      this.isSaving = false;
-    }
+    } catch (exception: any) {}
```
Comment (Wroud) [inline webapp/packages/core-ui/src/Form/FormState.ts:180]: please remove try/catch

File: webapp/packages/plugin-authentication-administration/src/Administration/Users/UserForm/Info/UserFormInfoPart.ts [hunk 5]
```diff
@@ -220,7 +211,7 @@ export class UserFormInfoPart extends FormPart<IUserFormInfoState, IUserFormStat
 
   protected override async loader() {
     let user: AdminUser | null = null;
-    const serverConfig = await this.serverConfigResource.load();
+    const [serverConfig, roles] = await Promise.all([this.serverConfigResource.load(), this.authRolesResource.load()]);
```
Comment (Wroud) [inline webapp/packages/plugin-authentication-administration/src/Administration/Users/UserForm/Info/UserFormInfoPart.ts]: please add `isOutdated()` checks for this resources to `isOutdated()` method

File: webapp/packages/plugin-authentication-administration/src/Administration/Users/UserForm/Info/UserFormInfoPart.ts [hunk 6]
```diff
@@ -138,14 +137,6 @@ export class UserFormInfoPart extends FormPart<IUserFormInfoState, IUserFormStat
...
     }
   }
-  protected override configure() {
-    const loadableStateContext = this.formState.dataContext.get(DATA_CONTEXT_LOADABLE_STATE)!;
-
-    loadableStateContext.getState('user-info', () => [
-      getCachedDataResourceLoaderState(this.serverConfigResource, () => undefined),
-      getCachedDataResourceLoaderState(this.authRolesResource, () => undefined),
-    ]);
-  }
```
Comment (Wroud) [inline webapp/packages/plugin-authentication-administration/src/Administration/Users/UserForm/Info/UserFormInfoPart.ts:148]: please check DATA_CONTEXT_LOADABLE_STATE usages and remove if needed

**Review by Wroud — CHANGES_REQUESTED** [review]

File: webapp/packages/core-ui/src/Form/FormState.ts (hunk 4 above)
Comment (sergeyteleshev) [reply webapp/packages/core-ui/src/Form/FormState.ts:180]: I don't think so
this is gonna break the logic in around 15 places

File: webapp/packages/plugin-authentication-administration/src/Administration/Users/UserForm/Info/UserFormInfoPart.ts (hunk 6 above)
Comment (sergeyteleshev) [reply webapp/packages/plugin-authentication-administration/src/Administration/Users/UserForm/Info/UserFormInfoPart.ts:148]: did not remove cause it is left in 1 place:
`/Users/stellech/development/dbeaver/cloudbeaver/webapp/packages/core-view/src/Menu/useMenuContext.ts`

File: webapp/packages/core-ui/src/Form/FormState.ts [hunk 7]
```diff
@@ -21,27 +21,30 @@ import { formStateContext } from './formStateContext';
 import type { IFormPart } from './IFormPart';
 import type { IFormState } from './IFormState';
 
+type PartsValues = IFormPart<any>[];
```
Comment (Wroud) [inline webapp/packages/core-ui/src/Form/FormState.ts]: this type is too simple, probably better to inline it, cuz usually you don't declare something like this:
```
type Strings = string[]
```

File: webapp/packages/core-ui/src/Form/FormState.ts (hunk 4 above)
Comment (Wroud) [reply webapp/packages/core-ui/src/Form/FormState.ts:180]: ah i see, you kept logic as before (not to throw) but now we have possibility of untracked exceptions

**Review by Wroud — CHANGES_REQUESTED** [review]

**Review by Wroud — APPROVED** [review]

**Pull request merged**